File: coding_test/avatar/game.cpp
#include <iostream>
#include <vector>
using namespace std;
int left_turn(int dirc)
{
    dirc -= 1;
    if (dirc == -1)
    {
        dirc = 3;
    }
    return dirc;
}

int main()
{
    int n, m, num, nx, ny;
    int x, y, dirc;
    int count = 1;
    int time_turn = 0;
    int arr[50][50];
    int d[50][50] = {
        0,
    };
    int dx[4] = {-1, 0, 1, 0};
    int dy[4] = {0, 1, 0, -1};

    cin >> n >> m;
    cin >> x >> y >> dirc;
    d[x][y] = 1;
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < m; j++)
        {
            cin >> num;
            arr[i][j] = num;
        }
    }

    while (1)
    {
        dirc = left_turn(dirc);
        nx = x + dx[dirc];
        ny = y + dy[dirc];
        if (arr[nx][ny] == 0 && d[nx][ny] == 0)
        {
            x = nx;
            y = ny;
            count += 1;
            time_turn = 0;
            d[nx][ny] = 1;
        }
        else
        {
            time_turn += 1;
        }
        if (time_turn == 4)
        {
            nx = x - dx[dirc];
            ny = y - dy[dirc];
            if (arr[nx][ny] == 0)
            {
                x = nx;
                y = ny;
            }
            else
            {
                break;
            }
            time_turn = 0;
        }
    }
    cout << count;
}

File: coding_test/avatar/game.py
n,m=map(int,input().split())
x,y,direction =map(int,input().split())
d=[[0]*m for _ in range(1,n+1)]
d[x][y] = 1
dx =[-1,0,1,0]
dy = [0,-1,0,1]
arr=[list(map(int,input().split())) for _ in range(n)]
def left_turn():
   global direction
   direction += 1
   if(direction == 4):
      direction = 0
count = 1
turn_time = 0
while 1:
   left_turn()
   nx = x+dx[direction]
   ny = y+dy[direction]
   if(d[nx][ny] == 0 and arr[nx][ny]==0):
      x,y=nx,ny
      d[x][y] = 1
      turn_time = 0
      count +=1
   else:
      turn_time += 1
   if(turn_time ==4):
      nx = x-dx[direction]
      ny = y-dy[direction]
      if(arr[nx][ny] ==0):
         x,y=nx,ny 
      else:
         break
      turn_time = 0

print(count)